pkg/virt-launcher/virtwrap/util/libvirt_helper.go

```diff
@@ -237,3 +251,124 @@ func NewDomain(dom cli.VirDomain) (*api.Domain, error) {
 	domain.GetObjectMeta().SetUID(domain.Spec.Metadata.KubeVirt.UID)
 	return domain, nil
 }
+
+// ForkAndMonitor itself to give qemu an extra grace period to properly terminate
+// in case of virt-launcher crashes
+func ForkAndMonitor(commandNamePrefix string) error {
+	cmd := exec.Command(os.Args[0], append(os.Args[1:], "--no-fork", "true")...)
+	stdout, err := cmd.StdoutPipe()
+	if err != nil {
+		return err
+	}
+	stderr, err := cmd.StderrPipe()
+	if err != nil {
+		return err
+	}
+
+	done := make(chan error, 2)
+	go func() {
+		io.Copy(os.Stdout, stdout)
+		done <- err
+	}()
+
+	go func() {
+		io.Copy(os.Stderr, stderr)
+		done <- err
+	}()
+	if err := cmd.Start(); err != nil {
+		log.Log.Reason(err).Error("failed to fork virt-launcher")
+		return err
+	}
+
+	sigs := make(chan os.Signal, 1)
+	signal.Notify(sigs, syscall.SIGINT, syscall.SIGTERM, syscall.SIGQUIT)
+	go func() {
+		for sig := range sigs {
+			log.Log.V(3).Log("signalling virt-launcher to shut down")
+			err := cmd.Process.Signal(syscall.SIGTERM)
+			if err != nil {
+				log.Log.Reason(err).Errorf("received signal %s but can't signal virt-launcher to shut down", sig.String())
+			}
+		}
+	}()
+
+	// stdout and stderr need to be fully read
+	<-done
+	<-done
+
+	if err := cmd.Wait(); err != nil {
+		exitCode := 1
+		if exiterr, ok := err.(*exec.ExitError); ok {
+			if status, ok := exiterr.Sys().(syscall.WaitStatus); ok {
+				exitCode = status.ExitStatus()
+			}
+		}
+		log.Log.Reason(err).Error("dirty virt-launcher shutdown")
+		pid, _ := virtlauncher.FindPid(commandNamePrefix)
+		if pid > 0 {
+			p, err := os.FindProcess(pid)
+			if err != nil {
+				return err
+			}
+			// Signal qemu to shutdown
+			err = p.Signal(syscall.SIGTERM)
+			if err != nil {
+				return err
+			}
+			// Wait for 10 seconds for the qemu process to disappear
+			err = wait.PollImmediate(1*time.Second, 10*time.Second, func() (bool, error) {
+				pid, _ := virtlauncher.FindPid(commandNamePrefix)
+				if pid == 0 {
+					return true, nil
+				}
+				return false, nil
+			})
+		}
+		os.Exit(exitCode)
+	}
+	return nil
+}
+
+func SetupLibvirt() error {
```

Comment:
The libvirt.sh script was doing something with default networks, but this function isn't. is that not necessary anymore?

Reply:
Not needed anymore since we are now explicitly connecting interfaces to prepared devices.